backend/app/notify/webhook.go

```diff
@@ -0,0 +1,143 @@
+package notify
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/umputun/remark42/backend/app/store"
+
+	log "github.com/go-pkgz/lgr"
+)
+
+const (
+	webhookTimeOut  = 5 * time.Second
+	webhookTemplate = `{"text": "{{.Text}}"}`
+)
+
+// WebhookParams contain settings for webhook notifications
+type WebhookParams struct {
+	WebhookURL string
+	Template   string
+	Headers    string
+	Timeout    time.Duration
+}
+
+// Webhook implements notify.Destination for Webhook notifications
+type Webhook struct {
+	WebhookParams
+}
+
+// NewWebhook makes Webhook
+func NewWebhook(params WebhookParams) (*Webhook, error) {
+	res := &Webhook{WebhookParams: params}
+	if res.Timeout == 0 {
+		res.Timeout = webhookTimeOut
+	}
+
+	if res.Template == "" {
+		res.Template = webhookTemplate
+	}
+
+	log.Printf("[DEBUG] create new Webhook notifier")
+
+	return res, nil
+}
+
+// Send to Slack channel
+func (t *Webhook) Send(ctx context.Context, req Request) error {
+	payload, err := buildWebhookPayload(req.Comment, t.Template)
+	if err != nil {
+		return errors.Wrap(err, "webhook notification failed")
+	}
+
+	err = t.doSend(ctx, &payload)
+	if err != nil {
+		return errors.Wrap(err, "webhook notification failed")
+	}
+
+	log.Printf("[DEBUG] send webhook notification, comment id %s", req.Comment.ID)
+
+	return nil
+}
+
+func buildWebhookPayload(comment store.Comment, tmpl string) (bytes.Buffer, error) {
+	var payload bytes.Buffer
+
+	payloadTmpl, err := template.New("webhook").Parse(tmpl)
+	if err != nil {
+		return payload, err
+	}
+
+	err = payloadTmpl.Execute(&payload, comment)
+	if err != nil {
+		return payload, err
+	}
+
+	return payload, nil
+}
+
+func (t *Webhook) doSend(ctx context.Context, payload io.Reader) error {
+	req, err := http.NewRequestWithContext(ctx, "POST", t.WebhookURL, payload)
+	if err != nil {
+		return errors.Wrap(err, "unable to create webhook request")
+	}
+
+	for h, hs := range buildHeaders(t.Headers) {
```

Comment:
I propose to pick full names for variables, now it's not clear unfortunately.

Reply:
Fixed